docs/standard/io/buffers.md

```diff
@@ -0,0 +1,188 @@
+
+- A fast path, that deals with the single segment case.
+- A slow path that deals with the data split across segments.
+
+There are a few approaches that can be used to process data in multi-segmented sequences:
+
+- Use the [`SequenceReader<T>`](#sequencereadert).
+- Parse data segment by segment, keeping track of the `SequencePosition` and index within the segment parsed. This avoids unnecessary allocations but may be inefficient, especially for small buffers.
+- Copy the `ReadOnlySequence<T>` to a contiguous array and treat it like a single buffer:
+  - If the `ReadOnlySequence<T>` has a length less than 256, it may be reasonable to copy the data into a stack-allocated buffer using the [stackalloc](../../csharp/language-reference/operators/stackalloc.md) operator.
+  - Copy the `ReadOnlySequence<T>` into a pooled array using <xref:System.Buffers.ArrayPool%601.Shared%2A?displayProperty=nameWithType>.
+  - Use [`ReadOnlySequence<T>.ToArray()`](xref:System.Buffers.BuffersExtensions.ToArray%2A). This isn't recommended in hot paths as it allocates a new `T[]` on the heap.
+
+The following examples demonstrate some common cases for processing `ReadOnlySequence<byte>`:
+
+##### Process binary data
+
+The following example parses a 4-byte big-endian integer length from the start of the `ReadOnlySequence<byte>`.
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet5)]
+
+##### Process text data
+
+The following example:
+
+- Finds the first newline (`\r\n`) in the `ReadOnlySequence<byte>` and returns it via the out 'line' parameter.
+- Trims that line, excluding the `\r\n` from the input buffer.
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet6)]
+
+##### Empty segments
+
+It's valid to store empty segments inside of a `ReadOnlySequence<T>`. Empty segments may occur while enumerating segments explicitly:
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet7)]
+
+The preceding code creates a `ReadOnlySequence<byte>` with empty segments and shows how those empty segments affect the various APIs:
+
+- `ReadOnlySequence<T>.Slice` with a `SequencePosition` pointing to an empty segment preserves that segment.
+- `ReadOnlySequence<T>.Slice` with an int skips over the empty segments.
+- Enumerating the `ReadOnlySequence<T>` enumerates the empty segments.
+
+### Potential problems with ReadOnlySequence\<T> and SequencePosition
+
+There are several unusual outcomes when dealing with a `ReadOnlySequence<T>`/`SequencePosition` vs. a normal `ReadOnlySpan<T>`/`ReadOnlyMemory<T>`/`T[]`/`int`:
+
+- `SequencePosition` is a position marker for a specific `ReadOnlySequence<T>`, not an absolute position. Because it's relative to a specific `ReadOnlySequence<T>`, it doesn't have meaning if used outside of the `ReadOnlySequence<T>` where it originated.
```

Comment:
This is a long sequence of text with no API links. It would be good to add when a new API is introduced.